### PR TITLE
Kh computer crud

### PR DESCRIPTION
**Pull request by KHardt**
## Link to Ticket
Closes : https://github.com/NSS-Therapeutic-Raccoons/WorkforceManagement/issues/8

## Description of Proposed Changes

Added Computer controller for list, create and delete. No edit
Added Computer Model
Added Computer Views: Create, Delete, Index and Details
Delete will not delete a computer if assigned to an employee
Added ComputerTest integration tests including input confirmation, delete and creating.
Note for testing purposes create computer will no longer have a date/time drop down and will require text input only.

## Steps to Test
Outline the steps to test

git fetch --all
git checkout KH-Computer-CRUD

Run server and test create , delete and details. 
Run computer test

## Impacted Areas in Application

List general components of the application that this PR will affect:

Computer create, delete, details and test

## Mentions @username

@figamus
@jeremiah3643
@ricky-bruner
@askingalot

## Definition of Done

1. The project must be fully documented. This includes the following:
    1. Complete README that documents the steps to install the code, how to install any dependencies, or system configuration needed.
    1. Every class must be documented with purpose, author, and methods.
    1. Every method must be documented with purpose and argument list - which itself must contain a short purpose for each argument.
1. The project must be able to run fully, and without errors, on each teammate's system.
1. Fulfills every requirement.
Every line of code has been peer reviewed.
1. For projects that require unit testing, core functionality must be identified and have at least one test for each.

### CODE REVIEW

**Review by ricky-bruner — APPROVED** [review]
I pulled it down, looked over the code, and ran all manual and integrated testing.

I dont like how delete works for computers, but it is technically what the ticket asked for. Great work klaus, stop being afraid of view models!

File: BangazonWorkforce/BangazonWorkforce.IntegrationTests/Config.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿
```
Comment (askingalot) [inline BangazonWorkforce/BangazonWorkforce.IntegrationTests/Config.cs:1]: This file shouldn't be here

File: BangazonWorkforce/BangazonWorkforce/Controllers/ComputerController.cs [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        [HttpPost, ActionName("Delete")]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> DeleteConfirmed(int id)
+        {
+            
+                using (IDbConnection conn = Connection)
+                {
+
+                    string sql = $@"DELETE FROM Computer WHERE id = {id}";
+                try {
```
Comment (askingalot) [inline BangazonWorkforce/BangazonWorkforce/Controllers/ComputerController.cs:124]: I'd love to see this code formatted better

File: BangazonWorkforce/BangazonWorkforce/Models/Computer.cs [hunk 3]
```diff
@@ -15,11 +14,9 @@ public class Computer
 
         public DateTime? DecomissionDate { get; set; }
 
+        public string Manufacturer { get; set; }
 
-        [Display(Name = "Assigned Computers")]
```
Comment (askingalot) [inline BangazonWorkforce/BangazonWorkforce/Models/Computer.cs:19]: What happened to the `make`?

**Review by jeremiah3643 — APPROVED** [review]
👍 other than Andy's comments. Everything looks good!

**Pull request merged**